tests/test_finaliser_safety.py

```diff
@@ -0,0 +1,104 @@
+# Tests for finaliser NULL safety in GIF and MJPEG modules
```

Comment:
If you want to add tests, please add them to `scripts/unittests/tests`. See the other unit tests for the style (basically a single function that returns True/False, also don't catch any exceptions the calling code already catches exceptions).